### DISCUSSION TITLE
Add possibility to export only a difficulty you're currently editing

### DISCUSSION THREAD

**Discussion by Skubers** [ppy/osu#24394] [Ideas]
If you wanted to send your GD or changes in your map to someone, you would usually send the .osu file of your diff from the Songs folder. You wouldn't use Export whole beatmap because the creator could've implemented some changes to other diffs and that would overwrite them.

There's no way to do that in lazer, so I have to do extra steps by exporting whole .osz package, extracting the diff and then sending it. That's a big hassle in the long run.

**Comment by alexpado** [comment]
This would also mean that a way to import the .osu file must be done too, as it would become an expectation to be able to import these after exporting

**Comment by peppy** [comment]
We will eventually implement a way to view all files in your beatmap. And probably reimplement "open .osu file" as per stable.

**Comment by nobbele** [comment]
(this is the most relevant discussion I could find on this issue)
@peppy Editing externally does exist now but it doesn't work for sending GDs to stable mappers.
If I may, I can make a PR for difficulty export like this, if it is something you wouldn't mind adding.
<img width="498" height="385" alt="image" src="https://github.com/user-attachments/assets/2be0011c-2161-4ab9-b1a5-8b12f518e823" />

**Reply by peppy** [reply]
Why doesn't it work?

**Reply by bdach** [reply]
@peppy It doesn't work because external mounting doesn't apply all of the stable workarounds of converting sliders to bezier or truncating all time instants everywhere to ints. The best you can currently do is export of entire mapset as compatibility -> rename .osz to .zip -> extract GD to send over.

I'd much rather just put in the work to add direct submission of guest difficulties to lazer. New BSS is already built to support it.

**Reply by peppy** [reply]
Right, that makes sense.
